Firestore/Source/Core/FSTFirestoreClient.mm

```diff
@@ -339,14 +340,15 @@ - (void)verifyNotShutdown {
   auto query_listener =
       QueryListener::Create(std::move(query), std::move(options), std::move(listener));
 
-  _workerQueue->Enqueue([self, query_listener] { [self.eventManager addListener:query_listener]; });
+  _workerQueue->Enqueue(
+      [self, query_listener] { _eventManager->AddQueryListener(query_listener); });
```

Comment:
Nit: you can move the `query_listener` (`_eventManager->AddQueryListener(std::move(query_listener));`).